e2e/tools/validator/src/validator/config/__init__.py

```diff
@@ -0,0 +1,73 @@
+# SPDX-FileCopyrightText: 2024-present Sunil Thaha <sthaha@redhat.com>
+#
+# SPDX-License-Identifier: APACHE-2.0
+
+import os
+import yaml
+from typing import NamedTuple
+
+class Remote(NamedTuple):
+    host: str
+    port: int
+    user: str
+    password: str
+    pkey: str
+
+    def __repr__(self):
+        return f"<Remote {self.user}@{self.host}>"
+
+class VM(NamedTuple):
+    pid: int
+
+class Metal(NamedTuple):
+    vm: VM
+
+class Prometheus(NamedTuple):
+    url: str
+
+class Validator(NamedTuple):
+    remote: Remote
+    metal: Metal
+    prometheus: Prometheus
+
+    def __repr__(self):
+        return f"<Config {self.remote}@{self.prometheus}>"
+
+
+def load(config_file: str) -> Validator:
+    """
+    Reads the YAML configuration file and returns a Config object.
+
+    Args:
+        config_file (str): Path to the YAML configuration file.
+
+    Returns:
+        Config: A named tuple containing the configuration values.
+    """
+    with open(config_file, 'r') as file:
+        config = yaml.safe_load(file)
+
+    remote_config = config['remote']
+    remote = Remote(
+        host=remote_config['host'],
+        port=remote_config.get('port', 22),
+        user=remote_config.get('username', 'fedora'),
+        password=remote_config.get('password', None),
+        pkey=os.path.expanduser(remote_config.get('pkey', '~/.ssh/id_rsa')),
```

Comment:
assuming null is passed to pkey, this will produce an error. This means for users that do not need to pass a pkey or want to use password cannot really set this to null.

Reply:
Good find! fixed now with a unittest